libs/game/mathUtil.ts

```diff
@@ -28,4 +28,62 @@ namespace Math {
         }
         return list[Math.randomRange(0, list.length - 1)];
     }
+
+    /**
+     * Fast, 16 bit, seedable (pseudo) random generator.
+     */
+    export class FastRandom {
+        // Implementation of the Galois Linear Feedback Shift Register
+        private lfsr: number;
+        // A value between 0x0001 and 0xFFFF to generate random values from
+        public seed: number;
+
+        /**
+         * Create a new Fast Random generator
+         * @param seed [Optional] initial seed between 0x0001 and 0xFFFF.
+         */
+        constructor(seed?: number) {
+            if (seed === undefined) seed = Math.randomRange(0x0001, 0xFFFF);
+            this.seed = seed;
+            this.lfsr = seed;
+        }
+
+        /**
+         * @returns the next random number between 0x0001 and 0xFFFF inclusive
+         */
+        next(): number {
+            return this.lfsr = (this.lfsr >> 1) ^ ((-(this.lfsr & 1)) & 0xb400);
+        }
+
+        /**
+         * @param min the minimum value to generate
+         * @param max the maximum value to generate
+         * @returns a random value between min and max (inclusive)
+         */
+        randomRange(min: number, max: number): number {
+            return min + this.next() % (max - min);
```

Comment:
only positive numbers right?

Reply:
if

```
min == -60
max == 0
this.next() == 42
```

then this will return ``-60 + 42 % (0 -  -60)`` which would be -18 which works, right?

I suppose I'm missing the invariant that ``min >= max``, though